src/BoundsInference.cpp

```diff
@@ -1200,44 +1202,53 @@ class BoundsInference : public IRMutator {
             // And the current bounds on its reduction variables, and
             // variables from extern for loops.
             if (producing >= 0) {
-                const Stage &s = stages[producing];
-                vector<string> vars;
-                if (s.func.has_extern_definition()) {
-                    vars = s.func.args();
-                }
-                if (stages[producing].stage > 0) {
-                    for (const ReductionVariable &rv : s.rvars) {
-                        vars.push_back(rv.var);
-                    }
-                }
-                for (const string &i : vars) {
-                    string var = s.stage_prefix + i;
-                    Interval in = bounds_of_inner_var(var, body);
-                    if (in.is_bounded()) {
-                        // bounds_of_inner_var doesn't understand
-                        // GuardWithIf, but we know split rvars never
-                        // have inner bounds that exceed the outer
-                        // ones.
-                        if (!s.rvars.empty()) {
-                            in.min = max(in.min, Variable::make(Int(32), var + ".min"));
-                            in.max = min(in.max, Variable::make(Int(32), var + ".max"));
+                for (const auto &fused : fused_with_f) {
+                    size_t si = 0;
```

Comment:
Please add a comment what stage index you're getting and why you need to do a search to find it. It's not clear to me what's happening.

Reply:
Added a comment, the logic there is the same as in the loop above where we add bounds for every function from the fused  group -- here it's the same except that it's for bounds of reduction variables.